Source/SwiftLintFramework/Models/Configuration.swift

```diff
@@ -213,6 +206,65 @@ public struct Configuration: Equatable {
     }
 }
 
+private func handleAlias(_ alias: String, ruleList: RuleList) -> String {
+    return ruleList.identifier(for: alias) ?? alias
+}
+
+private func defaultStringArray(_ object: Any?) -> [String] {
+    return [String].array(of: object) ?? []
+}
+
+private func validKeys(ruleList: RuleList) -> [String] {
+    return [
+        ConfigurationKey.disabledRules,
+        .enabledRules,
+        .excluded,
+        .included,
+        .optInRules,
+        .reporter,
+        .useNestedConfigs,
+        .warningThreshold,
+        .whitelistRules
+    ].map({ $0.rawValue }) + ruleList.allValidIdentifiers()
+}
+
+private func warnAboutDeprecations(_ dict: [String: Any],
+                                   disabledRules: [String] = [],
+                                   optInRules: [String] = [],
+                                   whitelistRules: [String] = [],
+                                   ruleList: RuleList) {
+
+    // Deprecation warning for "enabled_rules"
+    if dict[ConfigurationKey.enabledRules.rawValue] != nil {
+        queuedPrintError("'\(ConfigurationKey.enabledRules.rawValue)' has been renamed to " +
+            "'\(ConfigurationKey.optInRules.rawValue)' and will be completely removed in a " +
+            "future release.")
+    }
+
+    // Deprecation warning for "use_nested_configs"
+    if dict[ConfigurationKey.useNestedConfigs.rawValue] != nil {
+        queuedPrintError("Support for '\(ConfigurationKey.useNestedConfigs.rawValue)' has " +
+            "been deprecated and its value is now ignored. Nested configuration files are " +
+            "now always considered.")
+    }
+
+    // Deprecation warning for rules
+    let deprecatedRulesIdentifiers = ruleList.list.flatMap { (identifier, rule) -> [(String, String)] in
+        return rule.description.deprecatedAliases.map { ($0, identifier) }
+    }
+
+    let lists = Set(disabledRules + optInRules + whitelistRules)
```

Comment:
consider renaming the `lists` variable to something more informative, like `ruleIDs` or even `userProvidedRuleIDs`.